Migrate `verdi restapi` to use the click infrastructure

sphuber opened the pull request:
Fixes #1593

giovannipizzi commented on aiida/cmdline/commands/restapi.py:
```diff
@@ -8,56 +8,62 @@
     """
-    verdi command used to hook up the AiIDA REST API.
-    Example Usage:
+    Re-hash nodes filtered by identifier and or node class
```
wrong docstring here

DropD commented on aiida/cmdline/commands/restapi.py:
```diff
@@ -8,56 +8,62 @@
+
+
+@verdi_restapi.command('restapi', context_settings=CONTEXT_SETTINGS)
+@click.option('-h', '--host', type=click.STRING, default='127.0.0.1',
```
clashes with '-h' for help from CONTEXT_SETTINGS

sphuber commented:
I resolved the option clash and renamed `-h` to `-H`. I also added the `help_option_names` to the context settings directly in the `verdi` command group and removed it from all the sub commands. This works for all commands with sub commands, e.g. `verdi work list` will show `-h, --help`. However, for commands without sub commands, like `verdi restapi`, the default help string is used. I could not figure out how to do this. Does it maybe have to do with the fact that we are still in a half-way state between using click and argparse?
